lms/djangoapps/courseware/date_summary.py

```diff
@@ -402,21 +402,7 @@ def verified_upgrade_deadline_link(user, course=None, course_id=None):
     """
     if course is not None:
         course_id = course.id
-
-    ecommerce_service = EcommerceService()
-    if ecommerce_service.is_enabled(user):
-        course_mode = CourseMode.verified_mode_for_course(course_id)
-        if course_mode is not None:
-            return ecommerce_service.get_checkout_page_url(course_mode.sku)
-        else:
-            raise CourseModeNotFoundException('Cannot generate a verified upgrade link without a valid verified mode'
-                                              ' for course {}'.format(unicode(course_id)))
-    return reverse('verify_student_upgrade_and_verify', args=(course_id,))
-
-
-class CourseModeNotFoundException(Exception):
-    pass
-
+    return EcommerceService().upgrade_url(user, course_id)
```

Comment:
Did you just stop throwing the `CourseModeNotFound` exception, then? If so, are there callers of the `verified_upgrade_deadline_link` that are catching that and need to be updated to check for `None`?

Reply:
This exception was only called here and did not seem necessary as the EcommerceService  upgrade_url function does not throw an exception if it cannot find a link, it will just return none. Since these two functions are effectively the same I figured I was safe removing the exception. If you think we need it I can add one to the EcommerceService?